testhelper/test_options.go

```diff
@@ -71,6 +71,22 @@ type TestOptions struct {
 	WorkspaceName         string
 	WorkspacePath         string
 
+	// Use these options to specify a base terraform repo and branch to use for upgrade tests.
+	// If not supplied, the default logic will be used to determine the base repo and branch.
+	// Will be overidden by environment variables BASE_TERRAFORM_REPO and BASE_TERRAFORM_BRANCH if set.
+	BaseTerraformRepo   string
+	BaseTerraformBranch string
+
+	// BaseGitPersonalAccessToken specifies the Personal Access Token (PAT) to use for Git operations.
+	// This token provides authentication the base repository is private and accessed over HTTPS.
+	// If the environment variable GIT_PAT is set, it will override this value.
+	BaseGitPersonalAccessToken string
+
+	// BaseGitPrivateSshKey specifies the private SSH key to use for Git operations.
+	// This key provides authentication when the repository is private and accessed over SSH.
+	// If the environment variable SSH_PRIVATE_KEY is set, it will override this value.
+	BaseGitPrivateSshKey string
```

Comment:
This is incredibly risky for a user. I personally would never set either PAT or a private key in code. You could eliminate these two and just rely on ENV variables? would be a lot safer I think.